.ci/magician/github/membership_test.go

```diff
@@ -31,8 +33,76 @@ func TestTrustedContributors(t *testing.T) {
 
 func TestOnVacationReviewers(t *testing.T) {
 	for _, member := range onVacationReviewers {
-		if !slices.Contains(reviewerRotation, member) {
+		if !slices.Contains(reviewerRotation, member.id) {
 			t.Fatalf(`%v is not on reviewerRotation list`, member)
 		}
 	}
 }
+
+func TestAvailableReviewers(t *testing.T) {
+	tests := []struct {
+		name       string
+		rotation   []string
+		onVacation []onVacationReviewer
+		timeNow    time.Time
+		want       []string
+	}{
+		{
+			name:     "reviewers on vacation are excluded",
+			rotation: []string{"id1", "id2"},
+			onVacation: []onVacationReviewer{
+				{
+					id:        "id2",
+					startDate: time.Date(2024, 3, 29, 0, 0, 0, 0, time.UTC),
+					endDate:   time.Date(2024, 4, 2, 0, 0, 0, 0, time.UTC),
+				},
+			},
+			timeNow: time.Date(2024, 4, 1, 0, 0, 0, 0, time.UTC),
+			want:    []string{"id1"},
+		},
+		{
+			name:     "reviewers are included after vacation ends",
+			rotation: []string{"id1", "id2"},
+			onVacation: []onVacationReviewer{
+				{
+					id:        "id2",
+					startDate: time.Date(2024, 3, 29, 0, 0, 0, 0, time.UTC),
+					endDate:   time.Date(2024, 4, 2, 0, 0, 0, 0, time.UTC),
+				},
+			},
+			timeNow: time.Date(2024, 4, 2, 10, 0, 0, 0, time.UTC),
+			want:    []string{"id1", "id2"},
+		},
+		{
+			name:     "reviewers are included before vacation starts",
+			rotation: []string{"id1", "id2"},
+			onVacation: []onVacationReviewer{
+				{
+					id:        "id2",
+					startDate: time.Date(2024, 3, 29, 0, 0, 0, 0, time.UTC),
+					endDate:   time.Date(2024, 4, 2, 0, 0, 0, 0, time.UTC),
+				},
+			},
+			timeNow: time.Date(2024, 3, 28, 10, 0, 0, 0, time.UTC),
+			want:    []string{"id1", "id2"},
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			origRotation := reviewerRotation
+			origOnVacation := onVacationReviewers
+			reviewerRotation = test.rotation
+			onVacationReviewers = test.onVacation
+			defer func() {
```

Comment:
since AvailableReviewers now takes an explicit time argument the patching shouldn't be necessary.

Reply:
It was necessary because they are global vars. 
I refactor the new functions so revert the exported function back to its original signature, now this test only test the non exported function available().